src/SimpleGraphs/simpledigraph.jl

```diff
@@ -379,11 +379,23 @@ function copy(g::SimpleDiGraph{T}) where {T<:Integer}
     )
 end
 
-function ==(g::SimpleDiGraph, h::SimpleDiGraph)
-    return vertices(g) == vertices(h) &&
-           ne(g) == ne(h) &&
-           fadj(g) == fadj(h) &&
-           badj(g) == badj(h)
+function ==(g::SimpleDiGraph{T1}, h::SimpleDiGraph{T2}) where {T1,T2}
+    if (vertices(g) != vertices(h)) || (ne(g) != ne(h))
+        return false
+    end
+    fadjlist_g, fadjlist_h = fadj(g), fadj(h)
+    for v in eachindex(fadjlist_g, fadjlist_h)
+        if fadjlist_g[v] != fadjlist_h[v]
+            return false
+        end
+    end
+    badjlist_g, badjlist_h = badj(g), badj(h)
+    for v in eachindex(badjlist_g, badjlist_h)
+        if fadjlist_g[v] != fadjlist_h[v]
+            return false
+        end
+    end
+    return true
```

Comment:
Once https://github.com/JuliaRegistries/General/pull/86227 is merged and if you update JET version to v0.8.2, you will no longer need this change.

Reply:
Awesome, thanks!